local/owr_local_media_source.c

```diff
@@ -277,6 +319,9 @@ static gboolean shutdown_media_source(GHashTable *args)
     media_source = g_hash_table_lookup(args, "media_source");
     g_assert(media_source);
 
+    local_media_source = OWR_LOCAL_MEDIA_SOURCE(media_source);
+    g_clear_object(&local_media_source->priv->source_volume);
```

Comment:
To keep the reference counting clearer, it might make sense to hold a reference in `priv->source_volume`. So in addition to adding it to the bin, you would `gst_object_ref` it, and you would `gst_object_unref` it over here. This is quite minor, though, as you know the life time of the pipeline which means this isn't an actual problem.